backend/resources/product/products.controller.ts

```diff
@@ -199,4 +199,65 @@ export async function getManufacturers(req, res) {
         res.status(500).json({ message: "Server Error", error: error.message });
   }
 }
-  
\ No newline at end of file
+
+  //TODO #42 get products by stock amount endpoints @GaKa00 
+// get products by stock amount
+export async function getProductsbyStockAmount(req, res) {
+
+    const bool: 1 | -1 = parseInt(req.query.sort) === 1 ? 1 : -1;
+
+  try {
+    const result = await Product.aggregate([
+      {
+        $group: {
+          _id: "$product.name",
+         name: { $first: "$product.name" },
+          amountInStock: { $first: "$amountInStock" },
+        },
+      },
+    
+      bool ? {
+        $sort: { amountInStock: -1  }
+      } : { $sort: { amountInStock: 1 } }
+    ]);
+
+      const FilteredStockValue = result.map((Result) => ({
+        Product: Result.name,
+        totalAmount: Result.amountInStock,
+      }));
+      
+      res.status(200).json(FilteredStockValue);
+    
+  } catch (err) {
+    res.status(500).json({ message: "Server Error", error: err.message });
+  }
+}
+
+
+
+export async function getProductsbyCategory(req, res) {
+
+  try {
+    const result = await Product.aggregate([
+      {
+        $group: {
+          _id: "$name",
+          name: { $first: "$name" },
+          category: { $first: "$category" },
+        },
+      },
+
+      
+    ]);
+
+    const FilteredValue = result.map((Result) => ({
+      category:Result.category,
+      name: Result.name,
+      
+    }));
+
+    res.status(200).json(FilteredValue);
+  } catch (err) {
+    res.status(500).json({ message: "Server Error", error: err.message });
+  }
+}
```

Comment:
```suggestion
      bool ? {
        $sort: { amountInStock: -1  }
      } : { $sort: { amountInStock: 1 } }
```     { $sort: { amountInStock: -1  }}